smac/optimizer/smbo.py

```diff
@@ -347,3 +342,25 @@ def _get_timebound_for_intensification(self, time_spent):
                 time_spent, (1-frac_intensify), time_left, frac_intensify))
         return time_left
 
+
+class ChallengerList(object):
+
+    def __init__(self, challengers, configuration_space):
+        self.challengers = challengers
+        self.configuration_space = configuration_space
+        self._index = 0
+        self._next_is_random = False
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._index == len(self.challengers):
+            raise StopIteration
+        elif self._next_is_random:
```

Comment:
I fixed the first comment.

Regarding the second comment, we'd have to adapt intensify for that as well, or make sure that the second configuration is still random. Anyway, I think this is something for a second PR as this one is about increasing efficiency.